packages/adblocker/test/resources.test.ts

```diff
@@ -24,70 +24,97 @@ describe('#Resources', () => {
     expect(Resources.deserialize(buffer)).to.eql(resources);
   });
 
-  describe('#parse', () => {
-    it('parses empty resources', () => {
-      const resources = Resources.parse('', { checksum: 'checksum' });
-      expect(resources.checksum).to.equal('checksum');
-      expect(resources.js).to.eql(new Map());
-      expect(resources.resources).to.eql(new Map());
-    });
+  context('#parse', () => {
+    it('parses dependencies', () => {
+      const distribution: ResourcesDistribution = {
+        redirects: [],
+        scriptlets: [
+          {
+            names: ['a'],
+            content: 'function a() { b() }',
+            dependencies: ['b'],
+          },
+          {
+            names: ['b'],
+            content: 'function b() {}',
+            dependencies: [],
+          },
+        ],
+      };
+      const resources = Resources.parse(JSON.stringify(distribution), { checksum: '' });
 
-    it('parses one resource', () => {
-      const resources = Resources.parse('foo application/javascript\ncontent', {
-        checksum: 'checksum',
-      });
-      expect(resources.checksum).to.equal('checksum');
-      expect(resources.js).to.eql(new Map([['foo', 'content']]));
-      expect(resources.resources).to.eql(
-        new Map([['foo', { contentType: 'application/javascript', body: 'content' }]]),
+      expect(resources.getScriptlet('a')).to.be.eql(
+        wrapScriptletBody('function a() { b() }', ['function b() {}']),
       );
+      expect(resources.getScriptlet('b')).to.be.eql(wrapScriptletBody('function b() {}'));
     });
 
-    it('parses two resources', () => {
-      const resources = Resources.parse(
-        ['foo application/javascript\ncontent1', 'pixel.png image/png;base64\ncontent2'].join(
-          '\n\n',
-        ),
-        { checksum: 'checksum' },
-      );
-      expect(resources.checksum).to.equal('checksum');
-      expect(resources.js).to.eql(new Map([['foo', 'content1']]));
-      expect(resources.resources).to.eql(
-        new Map([
-          ['foo', { contentType: 'application/javascript', body: 'content1' }],
-          ['pixel.png', { contentType: 'image/png;base64', body: 'content2' }],
-        ]),
+    it('parses dependencies without function wrapper', () => {
+      const distribution: ResourcesDistribution = {
+        redirects: [],
+        scriptlets: [
+          {
+            names: ['a'],
+            content: 'function a() {}',
+            dependencies: ['b'],
+          },
+          {
+            names: ['b'],
+            content: 'function b() {}',
+            dependencies: [],
+          },
+        ],
+      };
+      const resources = Resources.parse(JSON.stringify(distribution), { checksum: '' });
+
+      expect(resources.getScriptlet('a')).to.be.eql(
```

Comment:
those tests should not use `getScriptlet` or `wrapScriptletBody` but check if `resources.resources` and `resources.scriplets` are in the right shape

Reply:
also, please add separate tests for `getScriptlet` and for `wrapScriptletBody`